src/matching_market.sol

```diff
@@ -106,8 +100,7 @@ contract MatchingMarket is MatchingEvents, ExpiringMarket, DSNote {
         returns (uint)
     {
         require(!locked, "Reentrancy attempt");
-        var fn = matchingEnabled ? _offeru : super.offer;
-        return fn(pay_amt, pay_gem, buy_amt, buy_gem);
+        return super.offer(pay_amt, pay_gem, buy_amt, buy_gem);
```

Comment:
should this be a noop? reverse? should we delete matchingEnabled flag? note that `_offeru` called super.offer anyway...

Reply:
Hmmm regarding this function, I think we could just remove it at all. As you said it doesn't do anything excepting checking for the reentrancy attempt. However the original `offer` function in `simple_market` has the `synchronized` modifier that already checks it. So yes IMO we should just remove the function.

Regarding `matchingEnabled`, at this point I'd say we also remove all regarding this. If the idea is making the contract simpler is a good step.

The only thing I'm concerned is we are working in this old version which doesn't even allow to compile with solc 0.5.